project/src/main/ui/career/career-win-world.gd

```diff
@@ -14,7 +14,34 @@ const ENVIRONMENT_PATH_BY_NAME := {
 	"marsh": "res://src/main/world/environment/marsh/MarshWinEnvironment.tscn",
 }
 
+func _ready() -> void:
+	if Engine.is_editor_hint():
+		return
+	
+	if PlayerData.career.current_region().sensei_absent:
+		var sensei := overworld_environment.find_creature(CreatureLibrary.SENSEI_ID)
+		sensei.visible = false
+	
+	_refresh_mood()
+
 
 func initial_environment_path() -> String:
 	var environment_name := PlayerData.career.current_region().overworld_environment_name
 	return ENVIRONMENT_PATH_BY_NAME.get(environment_name, DEFAULT_ENVIRONMENT_PATH)
+
+
+func _refresh_mood() -> void:
```

Comment:
Migrate comment,  `Updates the creature moods based on the player's performance`.

Reply:
Added comment